lldb/source/Target/TargetList.cpp

```diff
@@ -156,18 +157,15 @@ Status TargetList::CreateTargetInternal(
             }
           } else {
             // Only one arch and none was specified
-            prefer_platform_arch = true;
-            platform_arch = matching_module_spec.GetArchitecture();
+            update_platform_arch(matching_module_spec.GetArchitecture());
```

Comment:
Not sure about this. We are in the block were `platform_arch` is not valid, and you made it's replacement conditional on some of it's fields being unset.

Reply:
That's what the non-fat binary loading code does, too. I agree that it may have the desired behavior more by accident than by design.

Reply:
Sorry, but I think I double checked and this seems shady here. It's bot the same code as the current upstream. In this else block, `platform_arch` is not valid. `update_platform_arch` is going to update it conditionally on some of it's fields, but I'm not sure what that means give it is not valid.

Reply:
You are right. That's the same thing Jim found. I just kept answering your questions based on what I *meant* code to be, not what it actually was ...